device/src/region/mod.rs

```diff
@@ -113,19 +117,36 @@ macro_rules! region_dispatch {
   };
 }
 
-
 impl Configuration {
     pub fn new(region: Region) -> Configuration {
+        Configuration::with_state(State::new(region))
+    }
+
+    fn with_state(state: State) -> Configuration {
         Configuration {
-            state: State::new(region),
+            state,
+            receive_delay1: constants::RECEIVE_DELAY1,
+            receive_delay2: constants::RECEIVE_DELAY2,
+            join_accept_delay1: constants::JOIN_ACCEPT_DELAY1,
+            join_accept_delay2: constants::JOIN_ACCEPT_DELAY2,
         }
     }
-    pub(crate) fn create_tx_config(
-        &mut self,
-        random: u8,
-        datarate: DR,
-        frame: &Frame,
-    ) -> TxConfig {
+
+    // RECEIVE_DELAY2 is not configurable, LoRaWAN 1.0.3 Section 5.7: "The second reception slot opens one second 840after the first reception slot."
```

Comment:
nit: `840after` should at least be separated to `840 after`. Probably 840 is not necessary at all.